src/commands/generate/typeGenerator.ts

```diff
@@ -115,21 +119,32 @@ export async function generateEnum(definition: IEnumDefinition, simpleAst: Simpl
     const sourceFile = simpleAst.createSourceFile(definition.typeName);
 
     if (definition.values.length > 0) {
-        const namespaceDefinition = sourceFile.addNamespace({
+        const typeAliases = definition.values.map<TypeAliasDeclarationStructure>(enumValue => ({
+            kind: StructureKind.TypeAlias,
+            isExported: true,
+            name: enumValue.value,
+            type: doubleQuote(enumValue.value),
+            docs: addDeprecatedToDocs(enumValue),
+        }));
+        typeAliases.at(-1)!.trailingTrivia = `\n\n`;
+        const variableDeclarations = definition.values.map<VariableStatementStructure>(enumValue => ({
+            kind: StructureKind.VariableStatement,
+            isExported: true,
+            declarationKind: VariableDeclarationKind.Const,
+            declarations: [
+                {
+                    name: enumValue.value,
+                    initializer: `${doubleQuote(enumValue.value)} as ${doubleQuote(enumValue.value)}`,
+                },
+            ],
+        }));
+
+        const namespaceDefinition = sourceFile.addModule({
```

Comment:
The switch from `.addNamespace` to `.addModule` looks like a behavior change? Could we make a note of that in the PR description? Does this affect the API for downstream consumers?

I agree it's a bit weird to put constants in namespaces, but was expecting these improvements to happen in a separate PR from a `ts-simple-ast` to `ts-morph` migration.

Reply:
I think this is actually still a namespace, just different ts-morph API: https://github.com/palantir/conjure-typescript/pull/247/commits/ecdeaa4669a7503f98d0b3babb85bd3cdb5e13a2#diff-7b5b0ebe927145dbae5443002586ea7e6b9fbb6c410af93fc79f63b38b8260c4R144

relevant docs: https://ts-morph.com/details/modules

Reply:
Peter is right. Check the test output to see.

Reply:
Thanks for explaining @pgoldberg and @ericanderson! Happy I got this wrong.